javascript/0783-minimum-distance-between-bst-nodes.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Definition for a binary tree node.
+ * function TreeNode(val, left, right) {
+ *     this.val = (val===undefined ? 0 : val)
+ *     this.left = (left===undefined ? null : left)
+ *     this.right = (right===undefined ? null : right)
+ * }
+ */
+/**
+ * DFS (level order traversal)
+ * Time O(n) | Space O(n)
+ * https://leetcode.com/problems/minimum-distance-between-bst-nodes/
+ * @param {TreeNode} root
+ * @return {number}
+ */
+var minDiffInBST = function(root) {
+    
+    const sortedArr = [];
+
+    // levelOrderTraversal
+    const dfs = (node) => {
```

Comment:
Avoid nestled functions
```js

/**
 * Iterative - BST
 * Time O(N) | Space O(N)
 * https://leetcode.com/problems/minimum-distance-between-bst-nodes
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = (root, minDistance = Infinity) => {
    const inorderNodes = inorderTraversal(root);

    for (let i = 1; (i < inorderNodes.length); i++) {
        minDistance = Math.min(minDistance, (inorderNodes[i] - inorderNodes[(i - 1)]));
    }
    
    return minDistance;
}

var inorderTraversal = (root, inorderNodes = []) => {
    const isBaseCase = (root === null);
    if (isBaseCase) return inorderNodes;
    
    inorderTraversal(root.left, inorderNodes);
    inorderNodes.push(root.val);
    inorderTraversal(root.right, inorderNodes);

    return inorderNodes;
}

/**
 * DFS - BST
 * Time O(N) | Space O(N)
 * https://leetcode.com/problems/minimum-distance-between-bst-nodes
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = (root, prev = [ null ], minDistance = [ Infinity ]) => {
    const isBaseCase = (root === null);
    if (isBaseCase) return minDistance[0];

    minDiffInBST(root.left, prev, minDistance);

    const canUpdate = (prev[0] !== null);
    if (canUpdate) minDistance[0] = Math.min(minDistance[0], (root.val - prev[0].val));

    prev[0] = root;

    minDiffInBST(root.right, prev, minDistance);

    return minDistance[0];
}
```

Reply:
Omitting nested funciton.
Submission link for the proposed code: https://leetcode.com/problems/minimum-distance-between-bst-nodes/submissions/1388701576/

```javascript
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * DFS (level order traversal)
 * Time O(n) | Space O(n)
 * https://leetcode.com/problems/minimum-distance-between-bst-nodes/
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = function(root) {
    // levelOrderTraversal
    const sortedArr = dfs(root, []);
    
    let min = Infinity;
    for(let i = 1; i < sortedArr.length; i++) {
        min = Math.min(min, sortedArr[i] - sortedArr[i-1]);
    }
    return min;
};

const dfs = (node, sortedArr) => {
    if(!node) return;

    dfs(node.left, sortedArr);
    sortedArr.push(node.val)
    dfs(node.right, sortedArr);

    return sortedArr;
}
```

Reply:
@aakhtar3 (tagging to draw attention to the proposed code)

Reply:
@aadil42 For line 22, add spacing for the i - 1. Other than that the rest of the solution looks good 👍

Reply:
@Ykhan799, it's done.